### DISCUSSION TITLE
Pulsar Proxy cannot specify originalPrincipal

### DISCUSSION THREAD

**Discussion by lenglet-k** [apache/pulsar#23460] [Q&A]
Hello,

We want our JAVA client to communicate with our brokers via the pulsar proxy.
We use the biscuit authentication plugins: https://github.com/CleverCloud/biscuit-pulsar.

When the configuration is carried out without proxyRoles configured, we get the following error:
Illegal combination of role [biscuit:ROOT_BISCUIT] and originalPrincipal [biscuit:ATTENUATE_BISCUIT]: cannot specify originalPrincipal when connecting without valid proxy role

As soon as you put a proxyRoles in the broker configuration, the original principal is null? Should the proxyRole also be set in the proxy?

Thanks for your help

**Comment by Nowadays** [comment]
Also interested in this particular issue

**Comment by yuweisung** [comment]
Is this [description](https://pulsar.apache.org/docs/next/security-authorization/#proxy-roles) not clear?

**Reply by lenglet-k** [reply]
If that's clear, I'd misunderstood the principle.

**Comment by yuweisung** [comment]
proxy is a broker client. so you need to set authentication parameters in proxy.conf. the authentication parameters (client id) should match the proxyRole in broker.conf. 
Moreover, the real client role (original principal) should not be the same as proxyRole.

**Reply by lenglet-k** [reply]
Thanks for this answer. A last question, where is the documentation for know the permissions that we can apply on a role ? Because we want attenuate the proxyRole with the less privileges

**Comment by brunocascio** [comment]
I'm facing this issue only over websockets connection (http and pulsar protocols are fine)

```
 Illegal combination of role [pulsar-proxy-admin] and originalPrincipal [null]: originalPrincipal must be provided when connecting with a proxy role.
```
